store/postgres/src/chain_store.rs

```diff
@@ -1123,6 +1123,32 @@ mod data {
             }
         }
 
+        pub(super) fn delete_blocks_after(
+            &self,
+            conn: &mut PgConnection,
+            chain: &str,
+            block: i64,
+        ) -> Result<usize, Error> {
+            match self {
+                Storage::Shared => {
+                    use public::ethereum_blocks as b;
+
+                    diesel::delete(b::table)
+                        .filter(b::network_name.eq(chain))
+                        .filter(b::number.gt(block))
+                        .execute(conn)
+                        .map_err(Error::from)
+                }
+                Storage::Private(Schema { blocks, .. }) => {
```

Comment:
I think it does work because the ChainStore should already always have the right pool when it gets created. If you already tested that's great otherwise maybe rewind something in staging